### ISSUE TITLE
[REVIEW]: Mandyoc: A finite element code to simulate thermochemical convection in parallel

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#4070]
**Submitting author:** @victorsacek (<a href="http://orcid.org/0000-0001-9598-5081">Victor Sacek</a>)
**Repository:** https://github.com/ggciag/mandyoc
**Branch with paper.md** (empty if default branch): 
**Version:** v0.1.6
**Editor:** @jedbrown
**Reviewers:** @rbeucher, @psanan
**Archive:** 10.5281/zenodo.6390220

 

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/d41c910cbcd3f26e406ce9a72b94a5da/status.svg)](https://joss.theoj.org/papers/d41c910cbcd3f26e406ce9a72b94a5da)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/d41c910cbcd3f26e406ce9a72b94a5da"><img src="https://joss.theoj.org/papers/d41c910cbcd3f26e406ce9a72b94a5da/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/d41c910cbcd3f26e406ce9a72b94a5da/status.svg)](https://joss.theoj.org/papers/d41c910cbcd3f26e406ce9a72b94a5da)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/ggciag/mandyoc" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@rbeucher, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @jedbrown know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @rbeucher
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/ggciag/mandyoc">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ggciag/mandyoc">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@victorsacek) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] #4128
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @psanan
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/ggciag/mandyoc">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ggciag/mandyoc">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@victorsacek) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @rbeucher it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Wordcount for `paper.md` is 871

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.08 s (639.0 files/s, 154101.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
C++                             14           2302            350           5175
Python                           9            266            430            669
Markdown                         7            134              0            359
TeX                              2             34              0            342
C/C++ Header                     1             91             10            244
reStructuredText                 9            335            398            243
make                             2             20              6             76
YAML                             1              9             21             37
Bourne Shell                     2              4              0             10
CSS                              1              1              0              9
-------------------------------------------------------------------------------
SUM:                            48           3196           1215           7164
-------------------------------------------------------------------------------


Statistical information for the repository '21250cc553caf1aa2ba1642b' was
gathered on 2022/01/15.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Agustina                         4           109             67            1.28
Agustina Pesce                   1            17             10            0.20
Dave May                         1             2              0            0.01
Jamison Assunção                 3           716            700           10.26
Rafael Monteiro da S             3           391              4            2.86
Rafael Silva                     5           908             23            6.75
Victor Sacek                    69          9325           1524           78.64

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Agustina                    536          491.7          0.0                9.33
Dave May                      2          100.0         18.1                0.00
Jamison Assunção            637           89.0          3.0               23.86
Rafael Monteiro da S       1094          279.8          8.8                4.94
Rafael Silva                  1            0.1          0.5                0.00
Victor Sacek               7267           77.9          3.7                3.95
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1051/0004-6361/201629272 is OK
- 10.1051/0004-6361/201322068 is OK

MISSING DOIs

- 10.1111/j.1365-246x.2012.05388.x may be a valid DOI for title: A comparison of numerical surface topography calculations in geodynamic modelling: an evaluation of the ‘sticky air’method
- 10.1016/j.pepi.2010.04.007 may be a valid DOI for title: A stabilization algorithm for geodynamic numerical simulations with a free surface
- 10.1007/978-1-4612-1986-6_8 may be a valid DOI for title: Efficient Management of Parallelism in Object Oriented Numerical Software Libraries
- 10.1016/j.epsl.2016.11.026 may be a valid DOI for title: Post-rift influence of small-scale convection on the landscape evolution at divergent continental margins
- 10.1016/j.jog.2021.101830 may be a valid DOI for title: Lateral flow of thick continental lithospheric mantle during tectonic quiescence
- 10.1029/97jb01353 may be a valid DOI for title: A comparison of methods for the modeling of thermochemical convection

INVALID DOIs

- None
```

**Comment by jedbrown** [comment]
Hi @rbeucher! :wave: Welcome to JOSS and thanks for agreeing to review! The comments from @whedon above outline the review process, which takes place in this thread (possibly with issues filed in the Mandyoc repository). I'll be watching this thread if you have any questions.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention this issue so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within a month or so. Please let me know if you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me (@jedbrown) if you have any questions/concerns.

**Comment by jedbrown** [comment]
@victorsacek Please check out those missing DOI reported above :point_up:

**Comment by whedon** [comment]
:wave: @rbeucher, please update us on how your review is going (this is an automated reminder).

**Comment by rbeucher** [comment]
I'm starting the review today. Sorry for the delay.

**Comment by rbeucher** [comment]
@victorsacek 

Thanks for your submission. I have installed the code and I am currently running the examples. So far so good.
I have opened a series of issues on the repository issue tracker with some comments and suggestions on how to make the submission better.

The code compiles and run as described.
It needs some clearer examples that cover the range of functionalities describe in the documentation/paper.
The documentation is incomplete. It also needs simple tutorials to get the user started.
I would strongly suggest some basic automated tests....

The paper needs some work both in terms of structure and content. I think the content needs to be aligned with the
title. See my comments on the issue tracker.

**Comment by jedbrown** [comment]
@whedon add @psanan as reviewer

**Comment by whedon** [comment]
OK, @psanan is now a reviewer

**Comment by jedbrown** [comment]
Hi @psanan! :wave: Welcome to JOSS and thanks for agreeing to review! The comments from @whedon above outline the review process, which takes place in this thread (possibly with issues filed in the Mandyoc repository). I'll be watching this thread if you have any questions.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention this issue so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within a month or so. Please let me know if you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me (@jedbrown) if you have any questions/concerns.

**Comment by psanan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04070/joss.04070/10.21105.joss.04070.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04070/joss.04070/10.21105.joss.04070.pdf) :page_facing_up: :point_left:

**Comment by psanan** [comment]
@victorsacek, thanks for the submission! @rbeucher has opened several issues on the Mandyoc tracker, which cover the majority of points I wanted to make, at least generally, so I will just add my own comments to those issues where appropriate, and open new ones (limited in scope) if one of those issues is resolved but I still think some further change is required for acceptance.

**Comment by psanan** [comment]
@whedon generate pdf

**Comment by psanan** [comment]
@editorialbot generate pdf

**Comment by rafaelmds** [comment]
@rbeucher @psanan 

We published a new release of [Mandyoc v.0.1.5](https://github.com/ggciag/mandyoc/releases/tag/v0.1.5) addressing most of the review issues.

We kept issues open, since we made comments on each of them.

**Comment by rafaelmds** [comment]
@editorialbot generate pdf

**Comment by psanan** [comment]
@editorialbot generate pdf

**Comment by psanan** [comment]
@jedbrown @victorsacek I've checked off everything on my review list - looks ready from my perspective! Thanks again for the submission and the code - I particularly appreciate the attention paid to thorough documentation, including easy-to-follow, well-discussed, and reproducible examples covering both a benchmark case and a more realistic model.

**Comment by rbeucher** [comment]
@jedbrown @victorsacek I went through the changes and it looks ready to me. Thanks for doing the extra work!

**Comment by jedbrown** [comment]
Thanks for your diligent reviews @psanan @rbeucher.

@victorsacek Once https://github.com/ggciag/mandyoc/pull/83 is merged (or you've decided to revise otherwise to address these points), please tag a release ([annotated tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) preferred) and archive your release on Zenodo or similar, then report the DOI back here.

**Comment by jedbrown** [comment]
@editorialbot check references

**Comment by jedbrown** [comment]
:point_up: The PR addresses these.

**Comment by victorsacek** [comment]
Dear @jedbrown, @rbeucher, and @psanan

Thank you very much for the detailed revision of our code and documentation. 
Zenodo archived the version v0.1.5.1 of the Manydoc code:
10.5281/zenodo.6373350

**Comment by jedbrown** [comment]
@editorialbot set v0.1.5.1 as archive

**Comment by jedbrown** [comment]
@victorsacek Please edit the metadata on your DOI so the authors match this paper.

**Comment by jedbrown** [comment]
@editorialbot set 10.5281/zenodo.6373350 as archive

**Comment by jedbrown** [comment]
@editorialbot set v0.1.5.1 as version

**Comment by jedbrown** [comment]
@editorialbot generate pdf

**Comment by victorsacek** [comment]
> @victorsacek Please edit the metadata on your DOI so the authors match this paper.

Thank @jedbrown. Now the metadata was corrected.

**Comment by victorsacek** [comment]
Dear @jedbrown 

We observed that the tag for the last release is an invalid semver. We chose four numbers (v0.1.5.1) instead of three.
The correct version must be v0.1.6.
What is the best option to correct this mistake?

Thank you for your attention.

**Comment by jedbrown** [comment]
@victorsacek You can either tag v0.1.6 and update Zenodo or we can work with v0.1.5.1 (JOSS doesn't require that you use semver, strictly or otherwise). Your choice, just let us know which you choose.

**Comment by victorsacek** [comment]
Dear @jedbrown 
Thank you for your answer. We decided to update the tag to v0.1.6.
Here is the new Zenodo DOI: 10.5281/zenodo.6390220

**Comment by jedbrown** [comment]
@editorialbot set 10.5281/zenodo.6390220 as archive

**Comment by jedbrown** [comment]
@editorialbot recommend-accept

**Comment by kyleniemeyer** [comment]
@editorialbot accept

**Comment by kyleniemeyer** [comment]
Congratulations @victorsacek on your article's publication in JOSS!

Many thanks to @rbeucher and @psanan for reviewing this, and @jedbrown for editing.
